sycl/test/regression/bit_cast_unit32_to_half2.cpp

```diff
@@ -0,0 +1,7 @@
+// RUN: %clangxx -fsycl -fpreview-breaking-changes -c %s -o /dev/null
```

Comment:
```suggestion
// RUN: %clangxx -fsycl -fpreview-breaking-changes  %s -fsyntax-only
```

That said, why don't we actually verify the result?

Reply:
What to verify?
This test is intended to be just as a verification that compilation succeeds with bit_cast. Im sure there are other tests concerned with the functionality of bit_cast.

Reply:
> Im sure there are other tests concerned with the functionality of bit_cast

Then surely they would have caught the issue that this test targets...

Reply:
There has been a JRA ticket issued about this so the existing tests are missing this. The reason imo is that these tests so not have the preview-breaking-changes flag. Indeed, without that flag this test that im trying to submit fails.

Reply:
> why don't we actually verify the result?

What result do you mean? The test verifies the code returned by clang and stderr output.

`-fsyntax-only` skips parts of compilation process like LLVM IR generation and middle-end optimizations.

The problem we have with existing implementation can be detected with `-fsyntax-only`, but I don't object extending the scope.